src/robot/parsing/model/statements.py

```diff
@@ -64,6 +68,22 @@ def from_tokens(cls, tokens):
                 return handlers[token.type](tokens)
         return EmptyLine(tokens)
 
+    @classmethod
+    def from_params(cls, *args, **kwargs):
+        """Create statement from passed parameters. Required and optional arguments
+        should match class properties. Values are used to create matching tokens.
+
+        There is one notable difference for `Documentation` statement where
+        ``settings_header`` flag is used to determine if statement belongs to settings header
+        or test/keyword.
+
+        Most implementations support following general properties:
+        - `separator` whitespace inserted between each token. Default is ``'    '``.
+        - ``indent`` whitespace inserted before first token. Default is ``'    '``.
+        - ``eol`` end of line sign. Default is ``'\n'``.
```

Comment:
- Instead of `Default is '    '.` it would probably be better to say `Default is four spaces.`.
- `\n` in the docstring is a real newline. It should be changed to `\\n` or the docstring prefixed with `r`.